### ISSUE TITLE
UnicodeEncodeError in version: 2019.2.0

### ISSUE THREAD

**Issue by wuyudian1** [saltstack/salt#55162]
### Description of Issue
salt version: 2019.2.0
I wanna print some unicode strings in my python3 script, 
but `UnicodeEncodeError` encountered when I called that script by `salt-call --local cmd.run`
I can reproduce this problem by a simplified command:
```bash
salt-call --local cmd.run 'python3 -c "print(\"你好\")"'
```
### Setup
```bash
grep -Ev '^#|^$' /etc/salt/minion
master: xxxxxxx
master_port: xxxx
master_finger: xxxxxx
# I just changed these 3 fields
```

### Steps to Reproduce Issue
Command: `salt-call --local cmd.run 'python3 -c "print(\"你好\")"'`
Output: 
```python
[ERROR   ] Command 'python3 -c "print(\"你好\")"' failed with return code: 1
[ERROR   ] stdout: Unable to decode the command from the command line:
UnicodeEncodeError: 'utf-8' codec can't encode character '\udce4' in position 7: surrogates not allowed
[ERROR   ] retcode: 1
[ERROR   ] Command 'python3 -c "print(\"你好\")"' failed with return code: 1
[ERROR   ] output: Unable to decode the command from the command line:
UnicodeEncodeError: 'utf-8' codec can't encode character '\udce4' in position 7: surrogates not allowed
local:
    Unable to decode the command from the command line:
    UnicodeEncodeError: 'utf-8' codec can't encode character '\udce4' in position 7: surrogates not allowed
```
Command: `salt-call --local cmd.run 'python3 test.py'`
test.py:
```python
#!/usr/bin/env python3
#encoding=utf-8

print("你好")
```
Output:
```python
[ERROR   ] Command 'python3 test.py' failed with return code: 1
[ERROR   ] stdout: Traceback (most recent call last):
  File "test.py", line 4, in <module>
    print("\u4f60\u597d")
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-1: ordinal not in range(128)
[ERROR   ] retcode: 1
[ERROR   ] Command 'python3 test.py' failed with return code: 1
[ERROR   ] output: Traceback (most recent call last):
  File "test.py", line 4, in <module>
    print("\u4f60\u597d")
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-1: ordinal not in range(128)
```

### Versions Report
```bash
salt-call -V
Salt Version:
           Salt: 2019.2.0
 
Dependency Versions:
           cffi: 1.12.3
       cherrypy: Not Installed
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.2 (default, Oct  8 2019, 13:06:37)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1074-aws
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by frogunder** [comment]
@wuyudian1 Thank you for reporting this issue.

I am able to replicate it. Thanks.

**Comment by seandjones92** [comment]
Seeing a similar error in master logs, once this happens the master becomes unresponsive until restarted. I've not been able to reproduce it in this way but can achieve the interactive failure detailed in the original post.
```
2021-04-23 17:06:24,391 [salt.transport.zeromq                    :879 ][ERROR  ][4498] Some exception handling a payload from minion
Traceback (most recent call last):
 File "/usr/lib/python3.6/site-packages/salt/payload.py", line 196, in dumps
  msg, default=ext_type_encoder, use_bin_type=use_bin_type
 File "/usr/lib/python3.6/site-packages/salt/utils/msgpack.py", line 133, in packb
  return msgpack.packb(o, **_sanitize_msgpack_kwargs(kwargs))
 File "/usr/lib64/python3.6/site-packages/msgpack/__init__.py", line 46, in packb
  return Packer(**kwargs).pack(o)
 File "msgpack/_packer.pyx", line 284, in msgpack._cmsgpack.Packer.pack
 File "msgpack/_packer.pyx", line 290, in msgpack._cmsgpack.Packer.pack
 File "msgpack/_packer.pyx", line 287, in msgpack._cmsgpack.Packer.pack
 File "msgpack/_packer.pyx", line 234, in msgpack._cmsgpack.Packer._pack
 File "msgpack/_packer.pyx", line 263, in msgpack._cmsgpack.Packer._pack
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 17-18: surrogates not allowed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
 File "/usr/lib/python3.6/site-packages/salt/transport/zeromq.py", line 875, in handle_message
  ret, req_opts = yield self.payload_handler(payload)
 File "/usr/lib/python3.6/site-packages/salt/ext/tornado/gen.py", line 1056, in run
  value = future.result()
 File "/usr/lib/python3.6/site-packages/salt/ext/tornado/concurrent.py", line 249, in result
  raise_exc_info(self._exc_info)
 File "<string>", line 4, in raise_exc_info
 File "/usr/lib/python3.6/site-packages/salt/ext/tornado/gen.py", line 294, in wrapper
  result = func(*args, **kwargs)
 File "/usr/lib64/python3.6/types.py", line 248, in wrapped
  coro = func(*args, **kwargs)
 File "/usr/lib/python3.6/site-packages/salt/master.py", line 1060, in _handle_payload
  ret = {"aes": self._handle_aes, "clear": self._handle_clear}[key](load)
 File "/usr/lib/python3.6/site-packages/salt/master.py", line 1101, in _handle_clear
  ret = method(load), {"fun": "send_clear"}
 File "/usr/lib/python3.6/site-packages/salt/master.py", line 2286, in publish
  payload = self._prep_pub(minions, jid, clear_load, extra, missing)
 File "/usr/lib/python3.6/site-packages/salt/master.py", line 2373, in _prep_pub
  self.event.fire_event({"minions": minions}, clear_load["jid"])
 File "/usr/lib/python3.6/site-packages/salt/utils/event.py", line 797, in fire_event
  dump_data = self.serial.dumps(data, use_bin_type=True)
 File "/usr/lib/python3.6/site-packages/salt/payload.py", line 233, in dumps
  msg, default=ext_type_encoder, use_bin_type=use_bin_type
 File "/usr/lib/python3.6/site-packages/salt/utils/msgpack.py", line 133, in packb
  return msgpack.packb(o, **_sanitize_msgpack_kwargs(kwargs))
 File "/usr/lib64/python3.6/site-packages/msgpack/__init__.py", line 46, in packb
  return Packer(**kwargs).pack(o)
 File "msgpack/_packer.pyx", line 284, in msgpack._cmsgpack.Packer.pack
 File "msgpack/_packer.pyx", line 290, in msgpack._cmsgpack.Packer.pack
 File "msgpack/_packer.pyx", line 287, in msgpack._cmsgpack.Packer.pack
```
versions
```
# salt-call --versions
Salt Version:
          Salt: 3003

Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.7.3
     docker-py: Not Installed
         gitdb: 2.0.6
     gitpython: 3.0.7
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.6.1
        pygit2: Not Installed
        Python: 3.8.5 (default, Jan 27 2021, 15:41:15)
  python-gnupg: 0.4.5
        PyYAML: 5.3.1
         PyZMQ: 18.1.1
         smmap: 2.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.2

System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.72-microsoft-standard-WSL2
        system: Linux
       version: Ubuntu 20.04 focal
```

**Comment by dwoz** [comment]
@seandjones92 I think the issue you are seeing is distinctly different from @wuyudian1's issue. Please open separate issue for it.

@wuyudian1 I was able to reproduce this when running python 3.5. As a side note, it worked fine on both python 2.7 (2019.2.0) and python 3.7 (3003)

**Comment by dwoz** [comment]
This is an upstream python bug, and has been fixed in >=3.6

https://bugs.python.org/issue26919
https://bugs.python.org/issue26919

It can be re-produced outside of salt 
```
import subprocess                                                               
import sys                                                                      
                                                                                  
proc =  subprocess.Popen([sys.executable, '-c', 'print("你好")'], env={})                
proc.wait()
```
